app/controllers/hosts_controller.rb

```diff
@@ -454,6 +454,18 @@ def update_multiple_power_state
     redirect_back_or_to hosts_path
   end
 
+  def show_multiple_power_status
+    res = {}
+    if @hosts.respond_to? :each
+      @hosts.each do |host|
+        res[host.id] = host.supports_power_and_running? rescue false
+      end
+    else
+      logger.warn "Malformed host_ids parameter - not and array"
```

Comment:
I'm still not sure what condition will cause us to get `@hosts == nil`, since `find_multiple` should redirect in case of an error which will prevent this function from running (see http://guides.rubyonrails.org/action_controller_overview.html#filters) - if there is a condition that isn't caught correctly by the filter then we should fix it in the filter, not here.
if we do get here we will return a json with status 200, which is incorrect since there are errors.